### PR TITLE
307 New Version of Runtime Return Values Tutorial

### PR DESCRIPTION
**Pull request by ACStoneCL**
### Related links

[How-to guide on dealing with runtime return values #307](https://github.com/casper-network/docs/issues/307)

### Changes

This is a new version of the runtime return value tutorial, changing the example contract to a simple 'donation' contract, rather than the 'adder' used in the previous version.

It also includes the reorganization of the left navbar from #367 and the tutorial index/writing contract index changes from #362. As such, it should not be merged until after those two.

### Notes

Ran locally without issue.

### CODE REVIEW

File: source/docs/casper/dapp-dev-guide/tutorials/return-values-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+Contract code can optionally return a value to their immediate caller via `runtime::ret()`, whether that immediate caller is another contract code or session code. The returned value may be used within the context of the session or contract code, stored for later use, or discarded if not needed. Use of return values depends entirely on what the developer needs in that instance.
+
+Session code initiates actions on behalf of an account which is considered to be the caller. Therefore, it cannot return anything.
+
+## Contract Code {#return-contract-code}
+
+For example, if we create a contract to accept and keep a record of donations, we would use `runtime::ret()` to define the results that should be passed nto the caller as per the following:
```
Comment (ipopescu) [inline source/docs/casper/dapp-dev-guide/tutorials/return-values-tutorial.md:13]:
```suggestion
For example, if we create a contract to accept and keep a record of donations, we would use `runtime::ret()` to define the results that should be passed to the caller as per the following:
```

**Review by ipopescu — COMMENTED** [review]
@ACStoneCL is it possible to add a link to the full donation contract, if available in Git? 
The content looks great to me, but I'll leave the approval to @darthsiroftardis and @GloreenS.

File: source/docs/casper/dapp-dev-guide/tutorials/return-values-tutorial.md (hunk 1 above)
Comment (ACStoneCL) [reply source/docs/casper/dapp-dev-guide/tutorials/return-values-tutorial.md:13]: The donation contract is intended more as an example than an actual contract for someone to grab and use. I think it's better if we don't have it out and about for people to implement directly, but rather code their own if they want to implement a similar donation system.

**Comment by ACStoneCL** [comment]
Removed the 'do not merge' label as the necessary prerequisite pull requests have gone through.

**Review by darthsiroftardis — APPROVED** [review]
LGTM

**Pull request merged**